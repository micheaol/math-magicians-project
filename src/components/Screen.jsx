import PropTypes from 'prop-types';
import React from 'react';
import './Screen.css';

const Screen = ({ value }) =>
  (
    <div className="screen">
      <h1>{value}</h1>
    </div>
  );

Screen.propTypes = {
  value: PropTypes.node.isRequired,
};

export default Screen;
