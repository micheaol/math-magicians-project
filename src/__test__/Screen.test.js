import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Screen from '../components/Screen';

afterEach(() => {
  cleanup();
});
describe('render screen to the Dom', () => {
  test('render screen component to the screen', () => {
    render(<Screen />);
    const screenElement = screen.getByTestId('screen');
    expect(screenElement).toBeInTheDocument();
    expect(screenElement).toContainHTML('<h1></h1>');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Screen />);
    expect(tree).toMatchSnapshot();
  });
});
