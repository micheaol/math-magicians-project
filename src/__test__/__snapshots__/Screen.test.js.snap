// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render screen to the Dom matches snapshot 1`] = `
<div
  className="screen"
  data-testid="screen"
>
  <h1 />
</div>
`;
